Source/WebCore/css/parser/CSSPropertyParser.cpp

```diff
@@ -2771,6 +2771,45 @@ bool CSSPropertyParser::consumeWhiteSpaceShorthand(bool important)
     return true;
 }
 
+
+bool CSSPropertyParser::consumeAnimationRangeShorthand(bool important)
+{
+    // TODO: handle animation range sequences
+    RefPtr animationRangeStart = consumeAnimationRangeStart(m_range, m_context);
+    if (!animationRangeStart)
+        return false;
+
+    RefPtr<CSSValue> animationRangeEnd;
+    m_range.consumeWhitespace();
+    if (m_range.atEnd()) {
+        // From the spec: If <'animation-range-end'> is omitted and <'animation-range-start'> includes a component, then
+        // animation-range-end is set to that same and 100%. Otherwise, any omitted longhand is set to its initial value.
+
+        auto rangeEndValueForStartValue = [](const CSSValue& value) {
+            if (dynamicDowncast<CSSPrimitiveValue>(value) && SingleTimelineRange::isOffsetValue(*dynamicDowncast<CSSPrimitiveValue>(value)))
```

Comment:
The point of `dynamicDowncast<>` is to use the return value, not to call it twice, so please put it in a local variable. If you aren't using the return value you use `is<>` instead. And if you aren't checking the nullity you use `downcast<>` instead.

Reply:
Fixed.